src/camera.cpp

```diff
@@ -1044,6 +1065,18 @@ void CameraConfigDialogCtrlSet::OnSaturationChoiceChanged(wxCommandEvent& event)
     m_camSaturationADU->Enable(m_SaturationByADU->GetValue());
 }
 
+void CameraConfigDialogCtrlSet::OnSwBinningChecked(wxCommandEvent& event)
+{
+    wxArrayString opts;
+    int currBinning = GetIntChoice(m_binning, 1);
+    m_pCamera->GetBinningOpts(&opts, event.IsChecked());
+    FillChoiceItems(m_binning, opts);
+    if (event.IsChecked())
+        SetIntChoice(m_binning, currBinning);
+    else
+        SetIntChoice(m_binning, std::min(currBinning, (int) m_pCamera->MaxHwBinning));

```

Comment:
this is ok, but `wxMin` will also work without the cast to int

```suggestion
        SetIntChoice(m_binning, wxMin(currBinning, m_pCamera->MaxHwBinning));
```